Source/WebCore/loader/ContentFilter.h

```diff
@@ -89,6 +93,24 @@ class ContentFilter : public CanMakeWeakPtr<ContentFilter>, public CanMakeChecke
 private:
     using State = PlatformContentFilter::State;
 
+    class ContentFilterCallbackAggregator : public RefCounted<ContentFilterCallbackAggregator> {
```

Comment:
Are we confident that we don't need `ThreadSafeRefCounted` here? I did not look enough in the code but I know there is a background thread involved so I want to make sure this never gets ref'd / deref'd on different threads.

Reply:
`ThreadSafeRefCounted` should not be needed here, since `ContentFilterCallbackAggregator` is being captured in a completion handler on the main thread, and then the completion handler is moved to a background thread, before being moved back to the main thread again, and then finally called. However, to guard against future changes in this area, I think we can make it thread safe now.